src/framework/global/modularity/imoduleexport.h

```diff
@@ -45,6 +47,18 @@ struct IModuleExportCreator {
 };
 }
 
+#define DO_INTERFACE_ID(id, internal)                                           \
+public:                                                                         \
+    static const mu::modularity::InterfaceInfo& interfaceInfo() {               \
+        static constexpr std::string_view sig(IOC_FUNC_SIG);                    \
+        static const mu::modularity::InterfaceInfo info(#id, mu::modularity::moduleNameBySig(sig), internal);    \
```

Comment:
It looks like it might be possible to make `info` `static constexpr` (if you make the constructor of `InterfaceInfo` constrexpr too), or is it not? Not sure whether that would have an additional advantage.

Reply:
Made constexpr
Ideally, it would be good to catch errors at compile time (static assert for example)